common/flatpak-run.c

```diff
@@ -4167,6 +4170,104 @@ child_setup (gpointer user_data)
     fcntl (g_array_index (fd_array, int, i), F_SETFD, 0);
 }
 
+static gboolean
+forward_file (XdpDbusDocuments  *documents,
+              const char        *app_id,
+              const char        *file,
+              char             **out_doc_id,
+              GError           **error)
+{
+  int fd, fd_id;
+  g_autofree char *doc_id;
+  g_autoptr(GUnixFDList) fd_list = NULL;
+  const char *perms[] = { "read", "write", NULL };
+
```

Comment:
I think this code should just pass the argument as-is if the app already has access to the file in questions. That way we avoid creating a lot of weird  pathnames in the less sandboxed case, and we automatically get the full permissions on the pathnames.